### ISSUE TITLE
Pinned Items should be sortable

### ISSUE THREAD

**Issue by lemoness** [HabitRPG/habitica#9630]
You should be able to sort all items pinned to your Rewards column using drag-and-drop.

**Comment by kipraske** [comment]
I started doing a bit of work on this, though it is starting to get a bit more complicated than I expected. I will let you know if I run out of extra time to work on this ~

In particular `<draggable>` components really don't like being together under the same node at the same time. So in order to implement this it will be necessary to wrap the top elements in one component and the bottom elements in a separate component rather than the first component referencing the whole `.task-list` column element.

**Comment by lemoness** [comment]
@kipraske - Thanks so much for working on this! 

I've updated the labels on this ticket.

**Comment by kipraske** [comment]
I am making good progress on this bug. It has been definitely much more annoying than I anticipated. 

I am bringing this up now since next week is the holidays and I am likely going to ignore this for most of the week. So in case I die by eating too much figgy pudding or something next week here is a a working proof of concept of this bugfix: https://github.com/kipraske/habitica/tree/feature/sortable-reward-area 

It isn't ready for a PR yet since I have more testing to do, lots of code clean up, and probably a unit test or two on the new API endpoint that I added.

**Comment by kipraske** [comment]
I should make a status update now to this now. I have cleaned up the code and was ready to make a PR when I merged in the most recent develop branch changes. This adds the class-specific holiday equipment automatically to the rewards bar. Well it turns out that my code does not work at all with these seasonal pieces because internally the seasonal equipment does not use the isPinned flag like everything else does. It is always set to false. In fact, if you look at the seasonal equipment in the store, it will say it is unpinned even if it is.

Well, the real issue then is that the task view gets passed these seasonal items but the `user.pinnedItems` does not. So my drag and drop code, which updates the order of `user.pinnedItems` so it can be saved in the Database accidentally can save a temporary holiday item into `user.pinnedItems` where it throws and exception because these do not have isPinned set to true. As a result the entire rewards task column breaks for that user until you refresh the database back into a valid state.

So in short, I thought I was almost done, but now I am stuck because either:
 * I update my code so we do not store the pinned items in `user.pinnedItems` thus making the seasonal items always appear as the first 4 items in the list.
 * Change how the seasonal items work so they are the same place as the pinned items. This of course has the problem that then it will be tricky to make them disappear when they expire at the end of the season.

Neither of these options is so great so I am trying to think of a way that I can have my cake and eat it too here. The good news is that I have taken so long on this issue that I could test it with these seasonal items in there ~

**Comment by joedragons** [comment]
> This of course has the problem that then it will be tricky to make them disappear when they expire at the end of the season.

I may be over simplifying but can you borrow from the code that makes them disappear now from the Tasks page when not available?  Or are you saying pinning makes them persistent?  Just a thought.

**Comment by TheHollidayInn** [comment]
You may have to create a new array called pinnedItemsOrder that contains the pinned items ids sorted by the user. That would be the normalized form. I'd like to not at more info to the user, but I'm not sure of a better way

**Comment by kipraske** [comment]
@TheHollidayInn ~ Thanks for your help! That is how tasks are set up currently. I thought it was pretty weird, but you are totally right, I think that is worth a try.

**Comment by kipraske** [comment]
Alright another two weeks gone by, and I am back to being in good shape for this now using a new `pinnedItemsOrder` array to keep track of everything. Here is where I am at now:

* Everything is working like it should, and when you pin or unpin items, or change class to get a new set of seasonal items the order is preserved like one would expect.
* Any new item is placed at the top of the reward area
* I am not sure how to change the seasonal items to say, last fall or something to test what happens when they get automatically changed. It should work, but it isn't tested.
* There is an annoying bug I can't seem to figure out where when you drag the item the popover modal stays behind and prevents the user from dragging the item into position. There should be a way to propagate a message down to tell it to shut the modal on drag, but I don't seem to know enough about vue to do it in a smart way.
* I haven't written any tests, that should probably happen too since I added an API endpoint, but I have not learned the testing system yet and that will take time. I will be happy to write tests for this but I didn't know what sort of timeline you wanted this feature up and out by.
* I am getting busier at work now so I have less time to work on this. At the moment this isn't a problem, I am just slow, but let me know if you would like me to let someone pick up with what I have started to have this go faster.

If it helps to push up my current state into a branch for running those automated tests I will be happy to do that, otherwise the code is still on my fork linked above.

**Comment by kipraske** [comment]
Yay I figured out the popover bug. I hope I did it the right way. It turns out that vue does not work like react! Well that was the last major roadblock to getting feedback so I will open a pull request now.
